source/slang/slang-lower-to-ir.cpp

```diff
@@ -5995,29 +6028,17 @@ struct DeclLoweringVisitor : DeclVisitor<DeclLoweringVisitor, LoweredValInfo>
         return as<IRStringLit>(builder->getStringValue(stringLitExpr->value.getUnownedSlice()));
     }
 
-    LoweredValInfo lowerFuncDecl(FunctionDeclBase* decl)
+    void _lowerFuncResultAndParameterTypes(
+        ParameterLists& parameterLists,
+        List<IRType*>& paramTypes,
+        IRType*& irResultType,
+        IRBuilder* subBuilder,
+        IRGenContext* subContext,
+        FunctionDeclBase* decl)
```

Comment:
Interesting. My work on `property` declaration also ended up needing to refactor this operation to make some of the pieces be more reusable.

I have a feeling there is a painful merge ahead for me, but it seems like there is a good sign that the refactoring is something the codebase needed (since both of us seem to have made it).